core/src/test/scala/integration/kafka/api/PlaintextProducerSendTest.scala

```diff
@@ -120,17 +121,17 @@ class PlaintextProducerSendTest extends BaseProducerSendTest {
     }
   }
 
-  @ParameterizedTest(name = TestInfoUtils.TestWithParameterizedQuorumName)
```

Comment:
we are not running this test with both kraft and Zk by removing this. Could we please add it back?

Reply:
`quorum: String` this parameter was not used in the test. There are similar unit tests in this file which defines parameters like this @ValueSource(strings = Array("zk", "kraft")) but don't actually use them in the test. I have refrained from cleaning all the tests not to pollute this PR but that can be a quick follow up PR separately.

If there is some reason which I missed but we still want to the parametrized value, we can add it to the argument provider method like 

```
 Arguments.of("zk", TopicConfig.MESSAGE_TIMESTAMP_DIFFERENCE_MAX_MS_CONFIG, Long.box(System.currentTimeMillis() - fiveMinutesInMs)),
      Arguments.of("zk", TopicConfig.MESSAGE_TIMESTAMP_BEFORE_MAX_MS_CONFIG, Long.box(System.currentTimeMillis() - fiveMinutesInMs)),
      Arguments.of("zk", TopicConfig.MESSAGE_TIMESTAMP_AFTER_MAX_MS_CONFIG, Long.box(System.currentTimeMillis() + fiveMinutesInMs)),
 Arguments.of("kraft", TopicConfig.MESSAGE_TIMESTAMP_DIFFERENCE_MAX_MS_CONFIG, Long.box(System.currentTimeMillis() - fiveMinutesInMs)),
      Arguments.of("kraft", TopicConfig.MESSAGE_TIMESTAMP_BEFORE_MAX_MS_CONFIG, Long.box(System.currentTimeMillis() - fiveMinutesInMs)),
      Arguments.of("kraft", TopicConfig.MESSAGE_TIMESTAMP_AFTER_MAX_MS_CONFIG, Long.box(System.currentTimeMillis() + fiveMinutesInMs)),
```

Reply:
yeah, that part of the code is a but confusing. You see, we add an annotation `TestWithParameterizedQuorumName`. This annotation basically, changes the name of the test and adds quorum={0} to it. In addition `@ValueSource(strings = Array("zk", "kraft"))` sets the value of test name to be `testSendWithInvalidCreateTime.quorum=zk` or `testSendWithInvalidCreateTime.quorum=kraft`. After the test name has been modified at runtime, `TestInfoUtils.isKRaft` looks at the test name and determines if the test should be run as Zk test or a kraft test. That is why it may seem that this `quorum` param is unnecessary but actually it isn't.

Now, adding this to the argument provider should work, as long as it is the first argument.

Reply:
Thank you for clarifying that. I have added the `quorum` param and verified that the test name is actually updated during runtime as you described. Here is what I have captured from the test run out put
```
Gradle Test Run :core:test > Gradle Test Executor 3 > PlaintextProducerSendTest > testSendWithInvalidBeforeAndAfterTimestamp(String, String, long) > testSendWithInvalidBeforeAndAfterTimestamp(String, String, long).quorum=zk PASSED
Gradle Test Run :core:test > Gradle Test Executor 3 > PlaintextProducerSendTest > testSendWithInvalidBeforeAndAfterTimestamp(String, String, long) > testSendWithInvalidBeforeAndAfterTimestamp(String, String, long).quorum=zk PASSED
Gradle Test Run :core:test > Gradle Test Executor 3 > PlaintextProducerSendTest > testSendWithInvalidBeforeAndAfterTimestamp(String, String, long) > testSendWithInvalidBeforeAndAfterTimestamp(String, String, long).quorum=zk PASSED
Gradle Test Run :core:test > Gradle Test Executor 3 > PlaintextProducerSendTest > testSendWithInvalidBeforeAndAfterTimestamp(String, String, long) > testSendWithInvalidBeforeAndAfterTimestamp(String, String, long).quorum=kraft PASSED
Gradle Test Run :core:test > Gradle Test Executor 3 > PlaintextProducerSendTest > testSendWithInvalidBeforeAndAfterTimestamp(String, String, long) > testSendWithInvalidBeforeAndAfterTimestamp(String, String, long).quorum=kraft PASSED
Gradle Test Run :core:test > Gradle Test Executor 3 > PlaintextProducerSendTest > testSendWithInvalidBeforeAndAfterTimestamp(String, String, long) > testSendWithInvalidBeforeAndAfterTimestamp(String, String, long).quorum=kraft PASSED
```